### PR TITLE
Configure `iam.RolePolicy` resource and add example manifest

### PR DESCRIPTION
**Pull request by turkenf**
### Description of your changes

This PR includes configuring the `iam.RolePolicy` resource and adding the example manifest.

Related issue: https://github.com/upbound/provider-aws/issues/170, related PR: https://github.com/upbound/provider-aws/pull/745

I have:

- [x] Run `make reviewable test` to ensure this PR is ready for review.

### How has this code been tested

Manually and Uptest: https://github.com/upbound/provider-aws/actions/runs/6000658175
```
NAME                                READY   SYNCED   EXTERNAL-NAME   AGE
role.iam.aws.upbound.io/test-role   True    True     test-role       9m43s

NAME                                        READY   SYNCED   EXTERNAL-NAME           AGE
rolepolicy.iam.aws.upbound.io/test-policy   True    True     test-role:test_policy   9m43s
```

### CODE REVIEW

**Comment by turkenf** [comment]
/test-examples="examples/iam/rolepolicy.yaml"

**Comment by blakebarnett** [comment]
We're seeing performance problems with `provider-family-aws-iam` and this would be helpful to eliminate a bunch of managed resources. Any major reason it hasn't made it in yet?

**Comment by turkenf** [comment]
> We're seeing performance problems with provider-family-aws-iam and this would be helpful to eliminate a bunch of managed resources. Any major reason it hasn't made it in yet?

@blakebarnett, we expect the situation to become clearer this week([regarding this PR](https://github.com/upbound/provider-aws/pull/745)), but we cannot provide a definite date for now regarding when it will be finalized. We appreciate your patience and understanding during this process. If you have any further questions or concerns, please feel free to let us know.

**Comment by turkenf** [comment]
/test-examples="examples/iam/rolepolicy.yaml"

File: config/externalname.go [hunk 1]
```diff
@@ -486,10 +486,11 @@ var ExternalNameConfigs = map[string]config.ExternalName{
...
 	// arn:aws:iam::123456789012:policy/UsersManageOwnCredentials
-	"aws_iam_policy": iamPolicy(),
-	"aws_iam_user":   config.NameAsIdentifier,
-	"aws_iam_group":  config.NameAsIdentifier,
-	"aws_iam_role":   config.NameAsIdentifier,
+	"aws_iam_policy":      iamPolicy(),
+	"aws_iam_user":        config.NameAsIdentifier,
+	"aws_iam_group":       config.NameAsIdentifier,
+	"aws_iam_role":        config.NameAsIdentifier,
+	"aws_iam_role_policy": config.IdentifierFromProvider,
```
Comment (sergenyalcin) [inline config/externalname.go:493]: In registry [documentation](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/iam_role_policy#import), it seems that the id format is: `role_name:role_policy_name`.

Do you think we can construct it by using the `name` and `role` fields? Maybe you have already tried this, but I wanted to ask you anyway.


```suggestion
	"aws_iam_role_policy": config.TemplatedStringAsIdentifier("name", "{{ .parameters.role }}:{{ .external_name }}"),
```

**Review by sergenyalcin — COMMENTED** [review]
Thanks @turkenf I left a comment about external name configuration

**Comment by turkenf** [comment]
/test-examples="examples/iam/rolepolicy.yaml"

**Review by sergenyalcin — APPROVED** [review]
Thanks @turkenf LGTM!

**Pull request merged**